functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -11,166 +11,209 @@
+    cheServerURLBuilder.deleteCharAt(cheServerURLBuilder.lastIndexOf("&"));
+
+    Request request =
+        new Request.Builder()
+            .url(cheServerURLBuilder.toString())
+            .addHeader("Authorization", authorization)
+            .patch(body)
+            .build();
+
+    OkHttpClient client = new OkHttpClient.Builder().build();
+    Response response = client.newCall(request).execute();
+    if (response.isSuccessful()) {
+      relativePath = "workspace/" + WorkspaceID + "/runtime";
+      request =
+          new Request.Builder()
+              .url(testApiEndpointUrlProvider.get().toString().concat(relativePath))
+              .addHeader("Authorization", authorization)
+              .post(body)
+              .build();
+      response = client.newCall(request).execute();
       if (response.isSuccessful()) {
-        LOG.info("Prepare workspace request send. Starting workspace.");
-        sb = new StringBuilder("https://" + this.host);
-        sb.append("/api/workspace/");
-        sb.append(WorkspaceID);
-        sb.append("/runtime");
-        requestBuilder = new Request.Builder().url(sb.toString());
-        requestBuilder.addHeader("Authorization", "Bearer " + token);
-        request = requestBuilder.post(body).build();
-        response = client.newCall(request).execute();
-        if (response.isSuccessful()) {
-          LOG.info("Workspace was started. Waiting until workspace is running.");
-        }
+        LOG.info("Workspace was started successfully. Waiting for state RUNNING.");
+        return;
+      } else {
+        LOG.error("Che server start workspace failed:" + response.body().string());
       }
-    } catch (IOException e) {
-      LOG.error("Workspace start failed : " + e.getMessage(), e);
-      throw e;
+    } else {
+      LOG.error("Che starter start workspace failed:" + response.body().string());
     }
+    throw new RuntimeException("Workspace start failed.");
   }
 
-  private String getNameFromResponse(Response response) {
-    try {
-      String responseString = response.body().string();
-      Object jsonDocument =
-          Configuration.defaultConfiguration().jsonProvider().parse(responseString);
-      return JsonPath.read(jsonDocument, "$.config.name");
-    } catch (IOException e) {
-      LOG.error(e.getLocalizedMessage());
-      e.printStackTrace();
+  private String getNameFromResponse(HttpJsonResponse response)
+      throws RuntimeException, JsonParseException {
+    if (response.getResponseCode() == 200) {
+      String responseBody = response.asString();
+      JsonObject workspaceJsonObject = PARSER.parse(responseBody).getAsJsonObject();
+      JsonObject workspaceConfigJsonObject = workspaceJsonObject.get("config").getAsJsonObject();
+      return workspaceConfigJsonObject.get("name").getAsString();
+    } else {
+      String error = "Could not get name from response. Request failed or the response is empty.";
```

Comment:
Is there check that response is empty? Wouldn't it fail in line 199 if it is empty?

Reply:
response can be empty but it still contains a response code. (either 200, 404, 5xx)